src/common/module_builder.cc

```diff
@@ -65,6 +66,35 @@ ModuleBuilder::ModuleBuilder()
   m_context->appendDialectRegistry(registry);
 }
 
+static bool isScalarType(mlir::Type type) {
```

Comment:
Let's try to keep function declarations and definitions organized in top->down order in which they are first called, to facilitate easier code reading. If we have functions A and B, and A calls A1 and A2, I am fine with both `A, B, A1, A2` and `A, A1, A2, B` order. I usually group them closer together when they are locally called so I would choose `A, A1, A2, B`, but for example if A2 was called also from B (like `printModule` is called from many functions) then I would move it at the bottom.